### PR TITLE
Unify Task Event Registering

### PR DESCRIPTION
**Pull request by Spedwards**

### CODE REVIEW

**Comment by chobobobo** [comment]
This is a great step in removing repeated, copied code, will try to get it into canary.

**Comment by chobobobo** [comment]
Also merge https://github.com/cyberblast/screeps.ocs.internal/pull/276

File: task.js [hunk 1]
```diff
@@ -28,10 +28,29 @@ mod.flush = function () {
...
+        if (task.handleCreepDied) {
+            Creep.predictedRenewal.on(creep => task.handleCreepDied(creep.name));
+            Creep.died.on(task.handleCreepDied);
+        }
+        if (task.handleCreepError) Creep.error.on(task.handleCreepError);
+        // Room events
+        if (task.handleNewInvader) Room.newInvader.on(task.handleNewInvader);
+        if (task.handleKnownInvader) Room.knownInvader.on(task.handleKnownInvader);
+        if (task.handleGoneInvader) Room.goneInvader.on(task.handleGoneInvader);
+        if (task.handleRoomDied) Room.collapsed.on(task.handleRoomDied);
```
Comment (karlthepagan) [inline task.js]: This changes from a closure method into a method reference.

In other words you lose `this` functionality when registering a handler this way.

To restore this functionality change it to: `Room.collapsed.on(room => task.handleRoomDied(room));`

**Comment by chobobobo** [comment]
Once the recent changes have been tested in canary this should be merged

**Pull request merged**